webui/react/src/components/MetricSelectFilter.tsx

```diff
@@ -62,16 +87,91 @@ const MetricSelectFilter: React.FC<Props> = ({ metricNames, multiple, onChange,
     (onChange as MultipleHandler)(newMetric.sort(metricNameSorter));
   }, [ multiple, onChange, value ]);
 
+  const handleFiltering = useCallback((search: string, option) => {
```

Comment:
This doesn't need useCallback, does it?

Reply:
This one does because the callback is being sent to the child component of `SelectFilter`. If we don't use it for `handleFiltering`, anytime `MetricSelectFilter` re-renders, `SelectFilter` will re-render as well because `handleFiltering` will be recreated without the `useCallback`.

Reply:
Ah, that makes sense, thanks.